packages/swingset-liveslots/src/collectionManager.js

```diff
@@ -563,8 +581,9 @@ export function makeCollectionManager(
         doMoreGC =
           value.slots.map(vrm.removeReachableVref).some(b => b) || doMoreGC;
         syscall.vatstoreDelete(dbKey);
-        if (isEncodedRemotable(dbKey)) {
-          const keyVref = vrefFromDBKey(dbKey);
+        const encodedKey = dbKeyToEncodedKey(dbKey);
+        if (isEncodedRemotable(encodedKey)) {
+          const keyVref = vrefFromEncodedKey(encodedKey);
```

Comment:
Suggestion for a single-pass loop:
```js
export const collectionMetaKeys =
  new Set(['|entryCount', '|nextOrdinal', '|schemata']);
```
```js
      // visit every DB entry associated with the collection, which due to
      // sorting will be collection entries first and then a mixture of ordinal
      // mappings and size-independent metadata (both of which start with "|").
      for (const dbKey of enumerateKeysWithPrefix(syscall, dbKeyPrefix)) {
        const encodedKey = dbKeyToEncodedKey(dbKey);

        // preserve general metadata ("|entryCount" is cleared by our caller)
        if (collectionMetaKeys.has(encodedKey)) continue;

        if (!encodedKey.startsWith('|')) {
          // an entry; decref slots from its value
          const value = JSON.parse(syscall.vatstoreGet(dbKey));
          doMoreGC =
            value.slots.map(vrm.removeReachableVref).some(b => b) || doMoreGC;
        } else {
          // must be an ordinal mapping; decref its vref
          const keyVref = encodedKey.substring(1);
          parseVatSlot(keyVref);
          if (hasWeakKeys) {
            vrm.removeRecognizableVref(keyVref, `${collectionID}`, true);
          } else {
            doMoreGC = vrm.removeReachableVref(keyVref) || doMoreGC;
          }
        }

        // in either case, delete the DB entry
        syscall.vatstoreDelete(dbKey);
      }
```
or if you find the `enumerateKeysStartEnd` to be more clear, then a post-loop assertion that the collection no longer has any DB entries relating to its own entries (i.e., no entry data and no ordinal mapping data).
```js
      for (const dbKey of enumerateKeysWithPrefix(syscall, dbKeyPrefix)) {
        const encodedKey = dbKeyToEncodedKey(dbKey);
        collectionMetaKeys.has(encodedKey) ||
          Fail`internal: unexpected collection DB key suffix ${encodedKey}`;
      }
```

Reply:
yeah, ok, I've warmed to the single-pass loop, will apply

Reply:
I flipped the `if()` to make it simpler, please double-check my math

Reply:
FWIW, I preferred the other direction for the `if` as it represented better the order of iteration.